experimental/builder/include/ck_tile/builder/types.hpp

```diff
@@ -6,64 +6,91 @@
 #include <ostream>
 #include <string_view>
 #include <variant>
+#include <bit>
+#include <array>
 
 namespace ck_tile::builder {
 
 enum class DataType
 {
+    UNDEFINDED = 0,
     FP32,
     FP16,
     BF16,
     FP8,
+    INT32,
     I8,
     U8
 };
 
-// Memory layouts for 1D convolution tensors.
-// G: Group, N: Batch, K: Output Channel, C: Input Channel, W: Width
-// Enum defines Input, Weight, and Output tensor layouts respectively.
-enum class GroupConvLayout1D
+enum class TensorLayout
```

Comment:
This is pretty clean. Another strategy I sometimes see with flat lists of enums like this is to add extra enum values for start and end of ranges of grouped elements to enable classification checks with math. We can add that later if it helps logic, but concepts and lists with  `constexpr std::array<TensorLayout>` can also be used if we need more structure.